include/dr/shp/copy.hpp

```diff
@@ -105,6 +105,47 @@ OutputIt copy(InputIt first, InputIt last, OutputIt d_first) {
   return rv;
 }
 
+template <std::forward_iterator InputIt, lib::distributed_iterator OutputIt>
+cl::sycl::event copy_async(InputIt first, InputIt last, OutputIt d_first) {
```

Comment:
Once https://github.com/oneapi-src/distributed-ranges/pull/15 is merged please merge your branch to the latest main and add a test for this new _copy_async_ code.